src/js/components/DialogsComponent.jsx

```diff
@@ -15,137 +15,77 @@ var DialogsComponent = React.createClass({
 
-  onDialogPromptShow: function (message, defaultValue, dialogId, inputProps) {
-    this.setState({
-      dialog: {
-        type: DialogTypes.PROMPT,
-        message: message,
-        defaultValue: defaultValue,
-        inputProps: inputProps
-      },
-      currentId: dialogId
-    });
-  },
+    if (dialogData == null || currentDialogData == null) {
+      return;
+    }
 
-  onDialogClose: function (dialogId) {
-    if (dialogId !== this.state.currentId) {
-      return null;
+    if (currentDialogData.id !== dialogData.id) {
+      return;
     }
+
     this.setState({
-      dialog: null,
-      currentId: null
+      dialogData: null
     });
   },
 
-  handleAlertDismiss: function () {
-    DialogActions.alertDismiss(this.state.currentId);
-  },
-
-  handleConfirmDismiss: function () {
-    DialogActions.confirmDismiss(this.state.currentId);
-  },
-
-  handleConfirmAccept: function () {
-    DialogActions.confirmAccept(this.state.currentId);
-  },
-
-  handlePromptDismiss: function () {
-    DialogActions.promptDismiss(this.state.currentId);
+  handleAcceptDialog: function (value = null) {
+    DialogActions.acceptDialog(this.state.dialogData, value);
   },
 
-  handlePromptAccept: function (value) {
-    DialogActions.promptAccept(this.state.currentId, value);
+  handleDismissDialog: function () {
+    DialogActions.dismissDialog(this.state.dialogData);
   },
 
   getDialog: function () {
-    var dialog = this.state.dialog;
+    var dialogData = this.state.dialogData;
 
-    if (dialog == null) {
+    if (dialogData == null) {
       return null;
     }
 
-    switch (dialog.type) {
+    switch (dialogData.type) {
       case DialogTypes.ALERT:
         return (
-          <AlertModalComponent dismissButtonLabel={dialog.dismissButtonLabel}
-            message={dialog.message}
-            onDestroy={this.handleAlertDismiss} />
+          <AlertDialogComponent data={dialogData}
+              onAccept={this.handleAcceptDialog}
+              onDismiss={this.handleDismissDialog} />
         );
       case DialogTypes.CONFIRM:
         return (
-          <ConfirmModalComponent message={dialog.message}
-            onConfirm={this.handleConfirmAccept}
-            onDestroy={this.handleConfirmDismiss}
-            successButtonLabel={dialog.successButtonLabel} />
+          <ConfirmDialogComponent data={dialogData}
+              onAccept={this.handleAcceptDialog}
+              onDismiss={this.handleDismissDialog}  />
         );
       case DialogTypes.PROMPT:
         return (
-          <PromptModalComponent message={dialog.message}
-            defaultValue={dialog.defaultValue}
-            inputProps={dialog.inputProps}
-            onConfirm={this.handlePromptAccept}
-            onDestroy={this.handlePromptDismiss} />
+          <PromptDialogComponent data={dialogData}
+              onAccept={this.handleAcceptDialog}
+              onDismiss={this.handleDismissDialog}  />
```

Comment:
Honestly I'm still not sure how to indent jsx stuff. I thought it's +2, when wrapping attributes. Would you mind documenting this rule? https://github.com/mesosphere/marathon/issues/2910